modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/TableKeyDetails/index.js

```diff
@@ -0,0 +1,300 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayIcon from '@clayui/icon';
+import classNames from 'classnames';
+import getCurrentEndDate from '../../../../common/utils/getCurrentEndDate';
+import {
+	getFormatedProductName,
+	getProductDescription,
+	getProductName,
+	getStatusActivationTag,
+	hasVirtualCluster,
+} from '../../containers/DXPActivationKeysTable/utils/constants';
```

Comment:
I noticed that you create functions within constants folder.

Constants folder it's just for files with constants values as ROLES_TYPES, PAGES_TYPES, etc.
- For functions, need to be present in utils folder.
```suggestion
import {
	getFormatedProductName,
	getProductDescription,
	getProductName,
	getStatusActivationTag,
	hasVirtualCluster,
} from '../../containers/DXPActivationKeysTable/utils';
```